state/it/src/test/java/com/expediagroup/streamplatform/streamregistry/state/it/AgentIT.java

```diff
@@ -84,7 +86,29 @@ public void setUp() {
     kafkaEventSender = kafkaEventSender(topicName);
     entityView = new DefaultEntityView(kafkaEventReceiver(topicName, "groupId"));
     dummyAgent = new StoringEntityViewListener();
-    data = AgentData.generateData();
+    data = generateData();
+  }
+
+  @Test
+  public void concurrentModification() {
```

Comment:
@andykruth-ii This is the test for the `ConcurrentModificationException`. Locally, this would fail in less than 5 iterations  when using a `HashMap` (ran the test about 10 times). Never failed after the change to `ConcurrentHashMap`.